### PR TITLE
[N/A] Aligned build process with layouts service

### PR DESCRIPTION
**Pull request by pjbroadbent**
Demo app builds and runs via `npm install ; npm start`, with no additional build scripts.

The scripts and tooling to get this working are currently copy-pasted from Layouts service, to get things up and running quickly. Will find a cleaner way to share re-use these framework-level components going forward.

Also renames the small number of `CamelCase`-named folders to `camelCase`, as per existing conventions.

### CODE REVIEW

File: res/demo/app.json [hunk 1]
```diff
@@ -3,14 +3,14 @@
     "startup_app": {
         "name": "Notifications Demo",
         "description": "Demo of Openfin Notifications",
-        "url": "http://localhost:9048/demo/app-launcher.html",
```
Comment (tomrobertsOF) [inline res/demo/app.json]: Are we sure we want this to be 1337? That would prevent running local layouts and local notifs projects simultaneously.

File: src/provider/persistence/dataLayer/IDatastore.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {Entity} from '../../../shared/Models/Entity';
+import {Entity} from '../../../Shared/Models/Entity';
```
Comment (tomrobertsOF) [inline src/provider/persistence/dataLayer/IDatastore.ts]: You lower-cased the folder, but didn't update some of the references.

File: res/demo/app.json (hunk 1 above)
Comment (ryanpetersonOF) [reply res/demo/app.json]: +1 on this

File: src/provider/persistence/dataLayer/IDatastore.ts (hunk 2 above)
Comment (pjbroadbent) [reply src/provider/persistence/dataLayer/IDatastore.ts]: Oops. Have gone through and updated all the references now.

**Comment by ryanpetersonOF** [comment]
The module css stuff isn't fully working, but I think its better to get these changes out the door and worry about that later on.

**Pull request merged**